Corona-Warn-App/src/main/java/de/rki/coronawarnapp/appconfig/internal/ConfigDataContainer.kt

```diff
@@ -15,7 +15,14 @@ data class ConfigDataContainer(
 ) : ConfigData, ConfigMapping by mappedConfig {
 
     override val isDeviceTimeCorrect: Boolean
```

Comment:
What do you think about renaming this property to `isDeviceTimeCorrectOrAssumedCorrect` and add another property `isDeviceTimeCorrect` that simply checks `if(deviceTimeState ==ConfigData.DeviceTimeState.CORRECT)`. 

We need to check `isDeviceTimeCorrect` before performing data donation features.

Reply:
> What do you think about renaming this property to isDeviceTimeCorrectOrAssumedCorrect and add another property isDeviceTimeCorrect that simply checks if(deviceTimeState ==ConfigData.DeviceTimeState.CORRECT).

Maybe, changing the name or what it returns would be out of scope for this PR as it would affect many other features.

> We need to check isDeviceTimeCorrect before performing data donation features.

You don't need to do that manually. According to the tech spec the attestation should (and my implementation does?) check the device time and return a `SafetyNetException` if the device time is not sufficiently accurate.

API consumers (survey+analytics) should just need to call the classes in this PR, and only proceed if there was no exception.

Reply:
You are right, I missed that `CWASafetyNet` is doing the correct `deviceTime` checks. I thought it would call this method here.